pkg/vdbgen/vdb.go

```diff
@@ -475,6 +551,37 @@ func (d *DBGenerator) setSubclusterDetail(ctx context.Context) error {
 	return nil
 }
 
+// setImage will fetch the server version and use it to build
+// and set the image inside v.vdb.
+func (d *DBGenerator) setImage(ctx context.Context) error {
+	q := Queries[VersionQueryKey]
+	rows, err := d.Conn.QueryContext(ctx, q)
+	if err != nil {
+		return fmt.Errorf("failed running '%s': %w", q, err)
+	}
+	defer rows.Close()
+
+	if rows.Err() != nil {
+		return fmt.Errorf("failed running '%s': %w", q, rows.Err())
+	}
+	if !rows.Next() {
+		return errors.New("could not get Vertica version from meta-function")
+	}
+	var image string
+	if err := rows.Scan(&image); err != nil {
+		return fmt.Errorf("failed running '%s': %w", q, err)
+	}
+	// regex to match Vertica version
+	exp := regexp.MustCompile(`\d+(?:\.\d+){2}`)
```

Comment:
We should have a unit test for this.  Some sample inputs:
- Vertica Analytic Database 12.0.2-20221006
- Vertica Analytic Database 11.0.1-0

Reply:
I will add unit tests for the new  code.